lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/service/CriCheckingService.java

```diff
@@ -259,29 +260,40 @@ public JourneyResponse checkVcResponse(
         }
 
         if (configService.enabled(DL_AUTH_SOURCE_CHECK)
-                && getDcmawDlVc(newVcs).isPresent()
-                && getSuccessfulDlVc(sessionCredentials).isEmpty()) {
+                && requiresAuthoritativeSourceCheck(newVcs, sessionVcs)) {
             return JOURNEY_DL_AUTH_SOURCE_CHECK;
         }
 
         return JOURNEY_NEXT;
     }
 
-    private Optional<VerifiableCredential> getDcmawDlVc(List<VerifiableCredential> vcs) {
+    private boolean requiresAuthoritativeSourceCheck(
+            List<VerifiableCredential> newVcs, List<VerifiableCredential> sessionVcs) {
+        return extractDrivingPermitIdentifier(newVcs, Cri.DCMAW, false)
+                .map(
+                        identifier ->
+                                !identifier.equals(
+                                        extractDrivingPermitIdentifier(
+                                                        sessionVcs, Cri.DRIVING_LICENCE, true)
+                                                .orElse(null)))
+                .orElse(false);
+    }
+
+    private Optional<String> extractDrivingPermitIdentifier(
+            List<VerifiableCredential> vcs, Cri cri, boolean successfulCheck) {
         return vcs.stream()
-                .filter(vc -> Cri.DCMAW.equals(vc.getCri()))
-                .filter(
+                .filter(vc -> cri.equals(vc.getCri()))
+                .filter(successfulCheck ? VcHelper::isSuccessfulVc : x -> true)
```

Comment:
Do we need this condition? I don't think we ever care about unsuccessful VCs...

I suppose you could argue that if you already had a _failed_ DL VC for the same doc id then we should treat that as a failure here, but that seems too complicated - might as well just send them on the DL auth source check.

Reply:
Is the problem here not that if they had a failed DL VC from earlier in the journey that matched identifiers, by not checking that the succuess state for that VC, the default is to _not_ send them for an auth source check?

Reply:
Ah, I really meant the ternary as opposed to the success check itself - not that it matters now you've refactored :)